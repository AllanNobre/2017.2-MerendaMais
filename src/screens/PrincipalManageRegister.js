import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { TabViewAnimated, SceneMap, TabBar } from 'react-native-tab-view';
import Header from '../components/Header';
import ManageRegistersScreenContainer from '../Containers/ManageRegistersScreenContainer';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
const FirstRoute = () => <View style={[styles.container, { backgroundColor: '#ff4081' }]} />;
const ThirdRoute = () => <View style={[styles.container, { backgroundColor: '#673ab7' }]} />;

export default class Principal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      index: 0,
      routes: [
        { key: 'first', title: 'NÃ£o gerenciados' },
        { key: 'second', title: 'Validados' },
        { key: 'third', title: 'Excluidos' },
      ],
    };
  }

  handleIndexChange = index => this.setState({ index });

  renderFooter = props => <TabBar style={{ backgroundColor: '#FF9500', borderTopWidth: 1, borderBottomColor: 'black' }} tabStyle={{ paddingHorizontal: -0 }} {...props} />;

  renderHeader = props => <Header title={'Gerenciar Registro'} backButton {...props} />

  renderScene = SceneMap({
    first: ManageRegistersScreenContainer,
    second: FirstRoute,
    third: ThirdRoute,
  });

  render() {
    return (
      <TabViewAnimated
        style={styles.container}
        navigationState={this.state}
        renderHeader={this.renderHeader}
        renderFooter={this.renderFooter}
        renderScene={this.renderScene}
        onIndexChange={this.handleIndexChange}
      />
    );
  }
}
