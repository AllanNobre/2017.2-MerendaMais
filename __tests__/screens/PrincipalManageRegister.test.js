import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import configureStore from 'redux-mock-store';
import PrincipalManageRegister from '../../src/screens/PrincipalManageRegister';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing PrincipalManageRegister Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <PrincipalManageRegister />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
