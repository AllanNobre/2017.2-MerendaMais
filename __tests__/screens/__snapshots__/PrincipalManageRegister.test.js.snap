// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PrincipalManageRegister Screen renders as expected 1`] = `
<TabViewAnimated
  initialLayout={
    Object {
      "height": 0,
      "width": 0,
    }
  }
  navigationState={
    Object {
      "index": 0,
      "routes": Array [
        Object {
          "key": "first",
          "title": "NÃ£o gerenciados",
        },
        Object {
          "key": "second",
          "title": "Validados",
        },
      ],
    }
  }
  onIndexChange={[Function]}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderPager={[Function]}
  renderScene={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
